Tests/ClientTests/Wallpaper/WallpaperSelectorViewModelTests.swift

```diff
@@ -0,0 +1,134 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Glean
+import XCTest
+
+@testable import Client
+
+class WallpaperSelectorViewModelTests: XCTestCase {
+
+    private var wallpaperManager: WallpaperManagerInterface!
+
+    override func setUp() {
+        super.setUp()
+
+        wallpaperManager = WallpaperManagerMock()
+        addWallpaperCollections()
+
+        Glean.shared.resetGlean(clearStores: true)
+    }
+
+    override func tearDown() {
+        wallpaperManager = nil
+        super.tearDown()
+    }
+
+    func testInit_hasCorrectNumberOfWallpapers() {
+        let sut = createSut()
+        let expectedLayout: WallpaperSelectorViewModel.WallpaperSelectorLayout = .compact
+        XCTAssert(sut.sectionLayout == expectedLayout)
+        XCTAssert(sut.numberOfWallpapers == expectedLayout.maxItemsToDisplay)
+    }
+
+    func testUpdateSectionLayout_regularLayout_hasCorrectNumberOfWallpapers() {
+        let sut = createSut()
```

Comment:
There was no team consensus for this yet, but it was discussed that we could use `subject` instead in another PR. We can maybe chat about this so we all use the same convention.